tests/examples/migrations/remove_xhp_child_declarations.hack.expect

```diff
@@ -0,0 +1,23 @@
+class :no:children {
+}
+
+class :no-replacement {
+}
+
+class :with-validation-trait{
+  use XHPChildValidation;
+
+  protected function getChildrenDeclaration(): XHPChild\Constraint {}
+}
+
+class :without-trait{
+
+  // This is /probably/ nonsense, but can't be sure
+  protected function getChildrenDeclaration(): XHPChild\Constraint {}
+}
+
+class :with-comment {
+  // Blah blah
```

Comment:
We should probably skip classes that don't have the `XHPChildDeclarationConsistencyValidation` trait, i.e. the migration would only apply to classes where we know for sure that the children declaration is redundant. Otherwise we might be removing children declarations that don't have a valid replacement.

Reply:
I'd rather trust the user here - the usual flow should be:
- run `--add-xhp-child-declaration-methods` first
- test  thoroughly
- run `--add-xhp-child-declaration-methods` again, and check nothing's added
- run this on everything

> Otherwise we might be removing children declarations that don't have a valid replacement.

The foal is for children declarations to:
1. have no effect at runtime, i.e. be ignored by xhp-lib
2. be changed to parse errors

Removing unneeded is bad when we're at (2), but (1) needs  to happen first, and at that point it's a no-op